infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -576,69 +538,39 @@ public void processTopologyChangeNotification(DiscoveryServiceEvent event) {
         }
 
         if (isValid) {
-            if (clusterRoleChanged(discoveredTopology)) {
-                onClusterRoleChange(discoveredTopology);
-            } else {
-                onSinkClusterAddRemove(discoveredTopology);
-            }
+            onTopologyChange(discoveredTopology);
         } else {
             // Stop Log Replication in case this node was previously SOURCE but no longer belongs to the Topology
             stopLogReplication();
         }
     }
 
     /**
-     * Determine if there was a cluster role change between former topology and newly discovered
-     *
-     * @return true, cluster role changed
-     * false, otherwise
-     */
-    private boolean clusterRoleChanged(TopologyDescriptor discoveredTopology) {
-        // find the remote clusters which have been removed from remoteSourceClusters and added to remoteSinkClusters in
-        // the new topology.
-        Set<String> remoteSourceRemoved = Sets.difference(topologyDescriptor.getRemoteSourceClusters().keySet(),
-                discoveredTopology.getRemoteSourceClusters().keySet());
-        Set<String> remoteSinkAdded = Sets.difference(discoveredTopology.getRemoteSinkClusters().keySet(),
-                topologyDescriptor.getRemoteSinkClusters().keySet());
-        if (!Sets.intersection(remoteSinkAdded, remoteSourceRemoved).isEmpty()) {
-            return true;
-        }
-
-        // find the remote clusters which have been removed from remoteSinkClusters and added to remoteSourceClusters in
-        // the new topology
-        Set<String> remoteSinkRemoved = Sets.difference(topologyDescriptor.getRemoteSinkClusters().keySet(),
-                discoveredTopology.getRemoteSinkClusters().keySet());
-        Set<String> remoteSourceAdded = Sets.difference(discoveredTopology.getRemoteSourceClusters().keySet(),
-                topologyDescriptor.getRemoteSourceClusters().keySet());
-
-        return !Sets.intersection(remoteSourceAdded, remoteSinkRemoved).isEmpty();
-    }
-
-    /**
-     * Process a topology change where a sink cluster has been added or removed from the topology.
+     * Process a topology change where a remote cluster has been added or removed from the topology.
+     * (A role change is also treated as same)
      *
      * @param newTopology the new discovered topology
      */
-    // TODO V2: This method can be renamed to something more generic.  It is the default handling for any topology
-    //  change where the role has not changed.  This method assumes that only Sink clusters can change.  Once more
-    //  topologies are supported, Source clusters can also be added/removed.:::Coming in connectionModel PR
-    private void onSinkClusterAddRemove(TopologyDescriptor newTopology) {
-        log.debug("Sink Cluster has been added or removed");
+    private void onTopologyChange(TopologyDescriptor newTopology) {
+        log.debug("A role change or a remote cluster may have been added or removed");
```

Comment:
Shouldn't we add info log here to reflect topology change? Seems this is an important event for LR